[16-DOCS] 약속 제안 관련 스웨거 문서 작성

cpplovelove opened the pull request:
## 작업 목록
- [x] promisingController 스웨거 작성

## 세부 내용
- 약속 제안 관련 스웨거 문서 작성

## 논의 사항
- 이 전에 작성된 부분이 dateTime을 따로 가져오고 getPormisingInfo에서도 연관되어 있는 시간 정보를 가지고 있는데 이부분 수정하고 있습니다..! 스웨거 관련 부분 올려두고 수정해서 다시 올리겠습니다!

## 연결된 이슈
- #16

cpplovelove committed:
docs: promising controller 스웨거 세팅

-promising controller 스웨거 세팅

cpplovelove committed:
fix: 실수로 삭제한 부분 추가

cpplovelove committed:
refactor: promisingInfo date가져오는 형식 변경

- responseJson 형태로 가져오도록 수정

binimini commented on controllers/promising-controller.ts:
```diff
@@ -13,12 +22,18 @@ import timeUtil from '../utils/time';
   @Post('')
+  @ResponseSchema(PromisingRequest)
   @UseBefore(UserAuthMiddleware)
+  @ResponseSchema(CreatedPromisingResponse, { isArray: true })
```
ResponseSchema가 두갠데 의도된건가요?
응답 명시 확정해주세요~

binimini commented on controllers/promising-controller.ts:
```diff
@@ -40,19 +55,24 @@ class PromisingController {
-      availDates: availDates.map((date) => timeUtil.formatDate2String(date))
-    });
+    let availDates = await promisingDateService.findDatesById(promisingId);
+    const availDateList= availDates.map((date) => timeUtil.formatDate2String(date))
```
let 쓸 필요 없는 로직인데 사용하고 있는 거 같아요~

binimini commented on controllers/promising-controller.ts:
```diff
@@ -66,6 +86,9 @@ class PromisingController {
+  @OpenAPI({
+    summary: 'Get promisingList',
+    description: 'get promisingList by userId'  })
   @Get('/user')
```
요 부분 포함해서 ResponseSchema 없는 API들이 보이는데 응답이 명시 제대로 되는지 확인해주시겠어용??

binimini commented on controllers/promising-controller.ts:
```diff
@@ -74,6 +97,10 @@ class PromisingController {
+  @OpenAPI({
+    security: [{ bearerAuth: [] }],
+    summary: 'Time-response to promising',
+    description: 'time-table response to promising with promisingId : tableTable>date format= "2022-07-07" times foramt = [0,0,1,1,0,0,0,1,1,0] '  })
```
security 중복 명시되었어요 컨트롤러단에서 표시하고 있으니까 제거해주세요!

binimini commented on dtos/promising/request.ts:
```diff
@@ -10,14 +15,24 @@ class PromisingRequest {
   @IsNotEmpty()
   maxTime: Date;
   @IsNotEmpty()
+  @IsInt()
   categoryId: number;
   @IsNotEmpty()
   unit: number;
   @IsNotEmpty()
-  timeTable: Array<TimeOfDay>;
+  @IsArray()
```
IsInt처럼 자료형 명시해주면 IsNotEmpty 표시하지 않아도 필수 파라미터로 표시되는 걸로 알아요
확인해보시고 가능하면 제거해주시겠어요?

binimini commented on dtos/promising/request.ts:
```diff
@@ -10,14 +15,24 @@ class PromisingRequest {
   @IsNotEmpty()
-  availDate: Array<Date>;
-
+  @IsArray()
+  availDate: Date[];
```
ValidateNest 안해줘도 자료형 제대로 표시되나요? 확인 부탁드려요!

binimini commented on dtos/promising/request.ts:
```diff
@@ -30,8 +45,9 @@ class PromisingInfo {
   @IsNotEmpty()
   minTime: Date;
   @IsNotEmpty()
-  maxTime: Date;
+  maxTime: Date; 
```
IsNotEmpty보다는 IsString과 Matches(정규식)으로 Date 요구되는 형식 표시해주는게 나을꺼 같아요
제 PR에서 날짜 명시하는 부분 참고해주세요!

binimini reviewed (changes requested):
일단 이 부분 확인해주세요!! 작업이 밀려서 최대한 빨리 작업해주셔야할거같아요~
아니면 시간 안되시면 그냥 저한테 말씀해주시면 제가 수정하는 것도 괜찮을 거같아요! 🙃

cpplovelove committed:
fix: swagger 에러 픽스

-제대로 명시되지 않은 부분들 추가했습니다.

cpplovelove committed:
fix: promisingController swagger response 수정

- 잘못된 스웨거 응답 파트 수정

binimini committed:
refactor: model 내 class-validator 어노테이션 삭제

- validation 필요할시 sequelize-typescript 사용해서 추가할 예정

binimini committed:
refactor: DTO 및 응답 스키마 수정

- 중복으로 선언된 어노테이션 삭제
- TimeOfDay boolean[] 명시
- isArray 옵션 사용해 추가 DTO 사용하지 않도록 수정

binimini committed:
fix: DTO 통합 및 validation 어노테이션 수정

- 쟐못된 reference로 스웨거 문서에서 발생되던 에러 제거
- PromisingResponse 내에 avaliableDates 포함시켜 통일성 있게 처리 (멤버 추가 처리 필요)
- PromisingDateService Date[] 공통적으로 반환하도록 수정 및 DTO 생성자 처리
- Promising TimeTable 관련 반환 형식 상세 명시하도록 수정

binimini committed:
refactor: 스웨거 문서 에러 제거

- @JSONSchema() 가장 상단에 명시
- @IsArray() @Type() @ValidateNest({each:true}) 명시
- TimeTableResponse 변경

binimini committed:
refactor: API 설명 수정

binimini committed:
Merge branch 'dev' of https://github.com/YAPP-Github/20th-Android-Team-1-BE into feat-16-set-swagger

binimini reviewed (approved):
![스크린샷 2022-07-16 오전 10 48 36](https://user-images.githubusercontent.com/69030160/179340348-43535be6-c6ce-43cc-820f-14541927e66a.png)
요런식으로 에러나는 API가 꽤 있어서 타입 명시 어노테이션들 수정했습니다~ 🤣
수정하는 김에 DTO들 너무 제각각으로 따로 사용하고 있어서 최대한 통일성있게 리팩토링도 했어요!
(주된 변경사항은 Promising 내에서 availableDates 포함하도록 DTO 변경했습니다. 추가적인 사항은 커밋 설명 참고해주세요!)
`/api/promisings/user`의 API만 아직 리팩토링을 못했는데 오늘 뎁캠에서 할 예정입니당~
